client/test/router_store.spec.js

```diff
@@ -0,0 +1,669 @@
+    for (let i = 0; i < UtilConstants.MSGBYPAGE + 3; i++) {
+      const message = _.clone(fixtures.message1);
+      message.id = `id${i}`;
+      message.messageID = `meid${i}`;
+      message.conversationID = `c${i}`;
+      msgs.push(message);
+    }
+    return msgs;
+  }
+
+  /*
+   * Problem noticed:
+   *
+   * * FIXME: Private methods are melted with public ones.
+   * * FIXME: Store requests other stores without using getters.
+   * * FIXME: Some events require a value while it's never used
+   *          (ex: EDIT_ACCOUNT_REQUEST).
+   * * FIXME: EDIT_ACCOUNT_SUCCESS clears errors after creating some. Don't
+   *          know what to test.
+   * * FIXME: Code is shared bewteen client and server.
+   * * FIXME: getPreviousConversation and getNextConversation looks confusing.
+   */
+
+  before(() => {
+    dispatcher = new SpecDispatcher();
+    mockeryUtils.initDispatcher(dispatcher);
+    mockeryUtils.initForStores([
+      '../app/stores/router_store',
+      '../stores/account_store',
+      '../stores/message_store',
+      '../../../server/utils/constants',
+    ]);
+    routerStore = require('../app/stores/router_store');
+  });
+
+  describe('Actions', () => {
+    it('ROUTE_INITIALIZE', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.ADD_ACCOUNT_SUCCESS,
+        value: { account: _.clone(fixtures.account) },
+      });
+      dispatcher.dispatch({
+        type: ActionTypes.RECEIVE_RAW_MESSAGES,
+        value: [fixtures.message1, fixtures.message2],
+      });
+      dispatcher.dispatch({
+        type: ActionTypes.ROUTES_INITIALIZE,
+        value: fixtures.router,
+      });
+      assert.equal(routerStore.getRouter(), fixtures.router);
+    });
+    it('ROUTE_CHANGE', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.ROUTE_CHANGE,
+        value: {
+          accountID: fixtures.account.id,
+          mailboxID: fixtures.account.inboxMailbox,
+          tab: 'selected',
+          action: MessageActions.SHOW,
+          conversationID: fixtures.message1.conversationID,
+          messageID: fixtures.message1.id,
+          query: '',
+        },
+      });
+    });
+    it('ADD_ACCOUNT_REQUEST', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.ADD_ACCOUNT_REQUEST,
+        value: '',
+      });
+      assert.isTrue(routerStore.isWaiting());
+    });
+    it.skip('ADD_ACCOUNT_SUCCESS', (done) => {
+      dispatcher.dispatch({
+        type: ActionTypes.ADD_ACCOUNT_SUCCESS,
+        value: { account: _.clone(fixtures.account) },
+      });
+      setTimeout(() => {
+        assert.isFalse(routerStore.isWaiting());
+        assert.equal(routerStore.getAction(), MessageActions.SHOW_ALL);
+        assert.equal(routerStore.getAccountID(), fixtures.account.id);
+        assert.equal(routerStore.getMailboxID(), fixtures.account.inboxMailbox);
+        assert.equal(routerStore.getSelectedTab(), 'account');
+        assert.equal(currentURL, `#mailbox/${fixtures.account.inboxMailbox}`);
```

Comment:
Shouldnt you use this instead: `currentURL = _self.getCurrentURL isServer: false` ?